xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -969,6 +970,48 @@ void CGUIDialogPVRTimerSettings::RecordingGroupFiller(
     CLog::Log(LOGERROR, "CGUIDialogPVRTimerSettings::RecordingGroupFiller - No dialog");
 }
 
+void CGUIDialogPVRTimerSettings::MarginTimeFiller(
+  const CSetting *setting, std::vector< std::pair<std::string, int> > &list, int &current, void *data)
+{
+  CGUIDialogPVRTimerSettings *pThis = static_cast<CGUIDialogPVRTimerSettings*>(data);
+  if (pThis)
+  {
+    list.clear();
+
+    // Get global settings values
+    CPVRSettings::MarginTimeFiller(setting, list, current, data);
+
+    if (setting->GetId() == SETTING_TMR_BEGIN_PRE)
+      current = pThis->m_iMarginStart;
+    else
+      current = pThis->m_iMarginEnd;
+
+    bool bInsertValue = true;
+    auto it = list.begin();
+    while (it != list.end())
+    {
+      if (it->second == current)
+      {
+        bInsertValue = false;
+        break; // value already in list
+      }
+
+      if (it->second > current)
+        break;
+
+      ++it;
+    }
+
+    if (bInsertValue)
+    {
+      // PVR backend supplied value is not in the list of predefined values. Insert it.
+      list.insert(it, std::make_pair(StringUtils::Format(g_localizeStrings.Get(14044).c_str(), current) /* %i min */, current));
```

Comment:
As far as I know vector does not have a ctor taking a comparator.

Or do you mean to push_back the custom value after retrieving list from <code>CPVRSettings::MarginTimeFiller</code> and sorting the vector via std::sort after that? 

How are duplicates handled? I do not want duplicates in the list.